c7n/resources/glue.py

```diff
@@ -44,6 +44,29 @@ class ConnectionSubnetFilter(SubnetFilter):
     RelatedIdsExpression = 'PhysicalConnectionRequirements.SubnetId'
 
 
+@GlueConnection.filter_registry.register('route-table')
+class ConnectionRouteTableFilter(RelatedNetworkFilter):
+    """Filter Route Tables connected to a Glue Dev Endpoint
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: glue-connection-public-route
+            resource: aws.glue-connection
+            filters:
+                - type: route-table
+                  key: RouteTables[].Routes[].DestinationCidrBlock
+                  value: 0.0.0.0/0
+                  op: contains
+    """
+    permissions = ("ec2:DescribeRouteTables",)
+    association_type = "association.subnet-id"
+    describe_method = "describe_route_tables"
+    match_path = "PhysicalConnectionRequirements.SubnetId"
```

Comment:
does this work for a default route table?

Reply:
Hmm, since this relationship is implied, I guess not.. Will do some testing to confirm but might need to rethink this.

Reply:
Yep, confirmed that it would be blank when the subnet has that implied/default association with the main table. In this scenario, we will need to use the VpcId to identify the main table, then and go from there.

Reply:
I added a check for default route tables to handle this scenario. Adding filters for gateway type today that I think will get us most of the way there on this. I also will need to make it less generic and opinionated which is probably a good thing for now.